go/libkb/sig_chain.go

```diff
@@ -237,31 +239,35 @@ func (sc *SigChain) LoadFromServer(ctx context.Context, t *MerkleTriple, selfUID
 		},
 		NetContext: ctx,
 	})
-
 	if err != nil {
 		return
 	}
+	if finisher != nil {
+		defer finisher()
+	}
 
-	v := res.Body.AtKey("sigs")
-	var lim int
-	if lim, err = v.Len(); err != nil {
-		return
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
 	}
+	return sc.LoadServerBody(ctx, body, low, t, selfUID)
+}
 
+func (sc *SigChain) LoadServerBody(ctx context.Context, body []byte, low keybase1.Seqno, t *MerkleTriple, selfUID keybase1.UID) (dirtyTail *MerkleTriple, err error) {
 	foundTail := false
 
-	sc.G().Log.CDebugf(ctx, "| Got back %d new entries", lim)
-
 	var links ChainLinks
 	var tail *ChainLink
 
-	for i := 0; i < lim; i++ {
+	jsonparser.ArrayEach(body, func(value []byte, dataType jsonparser.ValueType, offset int, inErr error) {
+
 		var link *ChainLink
-		if link, err = ImportLinkFromServer(sc.G(), sc, v.AtIndex(i), selfUID); err != nil {
+		if link, err = ImportLinkFromServer(sc.G(), sc, value, selfUID); err != nil {
+			sc.G().Log.Debug("ImportLinkFromServer error: %s", err)
```

Comment:
I'd say as long as you're here, you should change to named `return`s --- pay down some tech debt.

Reply:
this is a return in the closure.